languages/csharp/BitwardenClient.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Newtonsoft.Json;
+
+namespace Bit.Sdk;
+
+public sealed class BitwardenClient : IDisposable
+{
+    private readonly ProjectsClient _projectsClient;
+    private readonly SecretsClient _secretsClient;
+    private readonly CommandRunner _commandRunner;
+    private readonly BitwardenSafeHandle _handle;
+
+    public BitwardenClient(BitwardenSettings? settings = null)
+    {
+        var clientSettings = new ClientSettings
+        {
+            ApiUrl = settings is { ApiUrl: not null } ? settings.ApiUrl : "https://api.bitwarden.com",
+            IdentityUrl = settings is { IdentityUrl: not null } ? settings.IdentityUrl : "https://identity.bitwarden.com",
+            DeviceType = DeviceType.Sdk,
+            UserAgent = "Bitwarden DOTNET-SDK"
+        };
+
+        var ptr = BitwardenLibrary.Init(clientSettings.ToJson());
+        _handle = new BitwardenSafeHandle(ptr);
+        _commandRunner = new CommandRunner(ptr);
```

Comment:
I wonder if this is a bit cumbersome, I don't really want anyone to use the `BitwardenLibrary` directly. Would it make sense to create an internal Client? And pass around that instead of the pointer and command runner.

```c#
class Client : IDisposable
{
    private readonly BitwardenSafeHandle _handle;

    internal Client(ClientSettings settings) {
        _handle = BitwardenLibrary.Init(settings.ToJson());
    }

    internal TReturn? RunCommand<TReturn>(Command command)
    {
        var req = command.ToJson();
        var result = BitwardenLibrary.RunCommand(req, _client);
        return JsonSerializer.Deserialize<TReturn>(result);
    }

    public void Dispose()
    {
        _handle.Dispose();
    }
}
```

Reply:
BitwardenLibrary  is internal, so accessible only within files in the same assembly. Also you can always use reflection  to get private methods. I don't think there is a need for this but if you prefer it this way I can do the change